src/App.tsx

```diff
@@ -15,31 +15,94 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+type State = {
+  selectedGood: string;
+};
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
+
+  deleteGood = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
+  addGood = (good: string) => {
+    this.setState({ selectedGood: good });
+  };
+
+  render() {
+    const title =
+      this.state.selectedGood === '' ? (
+        <h1 className="title">No goods selected</h1>
+      ) : (
+        <h1 className="title is-flex is-align-items-center">
+          {this.state.selectedGood} is selected
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={this.deleteGood}
+          />
+        </h1>
+      );
+
+    return (
+      <main className="section container">
+        {title}
+
+        <table className="table">
+          <tbody>
+            {goods.map(good => (
+              <tr
+                data-cy="Good"
+                key={good}
+                className={
+                  this.state.selectedGood === good
+                    ? 'has-background-success-light'
+                    : ''
+                }
+              >
+                <td>
+                  {this.state.selectedGood === good ? (
+                    <button
+                      data-cy="RemoveButton"
+                      type="button"
+                      className="button is-info"
+                      onClick={() => this.deleteGood()}
```

Comment:
[checklist](https://github.com/mate-academy/react_goods-selector/blob/master/checklist.md): If your method don't get any argument, you don't need to create new arrow function. Just pass your method.